CONTRIBUTING.md

```diff
@@ -1,176 +1,106 @@
-    you may not use this file except in compliance with the License.
-    You may obtain a copy of the License at
-
-        https://www.apache.org/licenses/LICENSE-2.0
-
-    Unless required by applicable law or agreed to in writing, software
-    distributed under the License is distributed on an "AS IS" BASIS,
-    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-    See the License for the specific language governing permissions and
-    limitations under the License.
-    """
+### Community Discussions
 
-    from typing import Any, Dict
-    from peekingduck.pipeline.nodes.node import AbstractNode
-
-    from peekingduck.pipeline.nodes.model.quack_utils.model import QuackModel
+Join us in our [discussion board](https://github.com/aimakerspace/PeekingDuck/discussions). Post your thread regarding
 
+- Questions on the project
+- Suggestions/features requests
+- Projects built using our project
 
-    class Node(AbstractNode):
-        """This node detects ducks in images."""
-        def __init__(self, config: Dict[str, Any]) -> None:
-            super().__init__(config, node_path=__name__)
+### Code Contributions
 
-            self.duck_thres = config["duck_thres"]
-            self.model = QuackModel(self.duck_thres)
-            self.logger.info("model loaded with configs: %s", self.duck_thres)
+Here is a guide of the general steps to contribute code to PeekingDuck.
 
+- Own an issue from the issues board. This is to prevent duplicate work.
 
-        def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
-            """Finds bounding boxes around ducks, if any.
+  1. Pick an issue from our [issues tracker](https://github.com/aimakerspace/PeekingDuck/issues) and indicate your interest to work on the issue.
+  2. If there are no one else working on the issue, the maintainer will assign the issue to you.
+  3. After receiving the confirmation from the maintainer, you may begin work on the issue.
 
-            Args:
-                inputs (dict): Dict with keys "img".
+- Contributing Code 
+  1. Do read our [code styles guidelines](#code-styles).
+  2. [Fork](https://docs.github.com/en/github/getting-started-with-github/quickstart/fork-a-repo) the aimakerspace/PeekingDuck repository. For more details in this process, Jake Jarvis has a [useful guide](https://jarv.is/notes/how-to-pull-request-fork-github/) that describes steps 2-6 and 10 in more detail.
+  3. Clone the forked branch to your local machine.
+  4. \[Recommended\] track the original repository as another remote. After which you will be able to receive updates using `git fetch`. This is useful for long term contributions to the repository.
+  5. In your local repository, create a descriptive branch to work on your issue.
+  6. Make the required changes within the branch.
+  7. Add tests (if possible).
+  8. Run the [test suite](#test-suites) and check that it passes.
+  9. Push your changes to github.
```

Comment:
To avoid confusion, might be good to emphasise that you're pushing to your own remote, not the official peekingduck one, so it won't appear as a branch there.

Reply:
Agreed. Added the changes